syntax = "proto3";

package tbMessages;

service TarboraMessages {
    rpc Connect(stream Message) returns (stream Message) {}
    rpc Subscribe(EventHeader) returns (Empty) {}
    rpc Desubscribe(EventHeader) returns (Empty) {}
}

enum MessageType {
    COMMAND = 0;
    EVENT = 1;
    LOCAL = 2;
    START_CONNECTION = 3;
    FINISH_CONNECTION = 4;
}

message Message {
    MessageType type = 1;
    uint32 from = 2;
    uint32 to = 3;
    string subject = 4;
    bytes body = 5;
}

message EventHeader {
    uint32 from = 1;
    string subject = 2;
}

message Empty {}
